test.js

```diff
@@ -512,6 +512,170 @@ test('Expose status route with additional route options, route schema options an
+    fastify.server.unref()
+    process.nextTick(() => block(500))
+    t.strictEqual(routeOptions.url, '/status-details')
+    t.strictEqual(routeOptions.logLevel, 'silent', 'log level not set')
+    t.deepEqual(routeOptions.config, customConfig, 'config not set')
+    fastify.close()
+  })
+
+  fastify.register(underPressure, {
+    exposeStatusDetailsRoute: {
+      routeOpts: {
+        logLevel: 'silent',
+        config: customConfig
+      },
+      url: '/status-details'
+    }
+  })
+
+  fastify.ready()
+})
+
+test('Expose status details route with additional route options and default url', t => {
+  t.plan(2)
+
+  const fastify = Fastify()
+
+  fastify.addHook('onRoute', (routeOptions) => {
+    fastify.server.unref()
+    process.nextTick(() => block(500))
+    t.strictEqual(routeOptions.url, '/status/details')
+    t.strictEqual(routeOptions.logLevel, 'silent', 'log level not set')
+    fastify.close()
+  })
+
+  fastify.register(underPressure, {
+    exposeStatusDetailsRoute: {
+      routeOpts: {
+        logLevel: 'silent'
+      }
+    }
+  })
+
+  fastify.ready()
+})
+
+test('Expose status details route with additional route options, route schema options', t => {
+  const routeSchemaOpts = { hide: true }
+
+  const fastify = Fastify()
+
+  fastify.addHook('onRoute', (routeOptions) => {
+    fastify.server.unref()
+    process.nextTick(() => block(500))
+    t.strictEqual(routeOptions.url, '/status/details')
+    t.strictEqual(routeOptions.logLevel, 'silent', 'log level not set')
+    t.deepEqual(routeOptions.schema, Object.assign({}, routeSchemaOpts, {
+      response: {
+        200: {
+          type: 'object',
+          properties: {
+            eventLoopDelay: { type: 'number' },
+            eventLoopUtilized: { type: 'number' },
+            heapUsed: { type: 'number' },
+            rssBytes: { type: 'number' }
+          }
+        }
+      }
+    }), 'config not set')
+    fastify.close()
+    t.end()
+  })
+
+  fastify.register(underPressure, {
+    exposeStatusDetailsRoute: {
+      routeOpts: {
+        logLevel: 'silent'
+      },
+      routeSchemaOpts,
+      url: '/status/details'
+    }
+  })
+
+  fastify.ready()
+})
+
+test('Expose status details route with additional route options, route schema options and default url', t => {
+  const routeSchemaOpts = { hide: true }
+
+  const fastify = Fastify()
+
+  fastify.addHook('onRoute', (routeOptions) => {
+    fastify.server.unref()
+    process.nextTick(() => block(500))
+    t.strictEqual(routeOptions.url, '/status/details')
+    t.strictEqual(routeOptions.logLevel, 'silent', 'log level not set')
+    t.deepEqual(routeOptions.schema, Object.assign({}, routeSchemaOpts, {
+      response: {
+        200: {
+          type: 'object',
+          properties: {
+            eventLoopDelay: { type: 'number' },
+            eventLoopUtilized: { type: 'number' },
+            heapUsed: { type: 'number' },
+            rssBytes: { type: 'number' }
```

Comment:
Could you a different schema instead of the default one?

Reply:
Not sure I follow your comment

Reply:
It is related to the previous comments (switch object.assign params)
As you said: this is correct!